src/components/Board/Board.js

```diff
@@ -31,28 +38,73 @@ const Board = ({ id }) => {
     setIsActiveForm(true);
   };
 
+  const handleOnDragEnd = (result) => {
+    const { destination, source, draggableId } = result;
+
+    if (!destination) {
+      return;
+    }
+
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return;
+    }
+
+    dispatch(
+      updateCardColumnID({
+        id: draggableId,
+        columnID: destination.droppableId,
+      })
+    );
+
+    dispatch(
+      updateCardsIndexOnDragEnd({
+        id: draggableId,
+        columnID: destination.droppableId,
+        index: destination.index,
+      })
+    );
+  };
+
+  const handleOnDragStart = (result) => {
+    const { draggableId } = result;
+
+    dispatch(
+      updateCardsIndexOnDragStart({
+        id: draggableId,
+      })
+    );
+  };
+
   return (
-    <div className={styles.container}>
-      {columns.map((column) => (
-        <Column
-          key={column.id}
-          column={{ id: column.id, title: column.title }}
-        />
-      ))}
-      {isActiveForm ? (
-        <AddForm
-          buttonText="Add new Column"
-          submitAction={(text) => submitAction(text)}
-          placeholderText="Type title for new column"
-          cancelAction={() => cancelAction()}
-        />
-      ) : (
-        <button className={styles["add-new-column"]} onClick={handleClick}>
-          <FontAwesomeIcon className={styles.icon} icon={faPlus} />
-          Add new
-        </button>
-      )}
-    </div>
+    <DragDropContext
+      onDragEnd={handleOnDragEnd}
+      onDragStart={handleOnDragStart}
+    >
+      <div className={styles.container}>
+        {columns.map((column) => (
+          <Column
+            key={column.id}
+            column={{ id: column.id, title: column.title }}
+          />
+        ))}
+        {isActiveForm ? (
+          <AddForm
+            buttonText="Add new Column"
+            submitAction={(text) => submitAction(text)}
```

Comment:
```suggestion
            submitAction={submitAction}
```
That should work the same.